test/specs/rules/require-template-strings.js

```diff
@@ -7,86 +7,155 @@ describe('rules/require-template-strings', function() {
     beforeEach(function() {
         checker = new Checker();
         checker.registerDefaultRules();
-        checker.configure({ esnext: true, requireTemplateStrings: true });
     });
 
-    it('should report the use of string concatenation with a identifier on the left', function() {
-        assert(checker.checkString('a + "a"').getErrorCount() === 1);
+    function errorChecks(string) {
+        it('should report the use of string concatenation with a identifier on the left', function() {
+            assert(checker.checkString('a + ' + string).getErrorCount() === 1);
+        });
+
+        it('should report the use of string concatenation with a identifier on the right', function() {
+            assert(checker.checkString(string + ' + a').getErrorCount() === 1);
+        });
+
+        it('should report the use of string concatenation with right handed binary expression',
+            function() {
+                assert(checker.checkString('"test" + (a + b)').getErrorCount() === 1);
+            }
+        );
+
+        it('should report the use of string concatenation with left handed binary expression',
+            function() {
+                assert(checker.checkString('(a + b) + "test"').getErrorCount() === 1);
```

Comment:
Non-blocker: prefer `assert.equal` instead of a `===` usage for better error messages when this test fails.